components/automate-ui/src/app/modules/infra-proxy/create-environment-modal/create-environment-modal.component.spec.ts

```diff
@@ -0,0 +1,177 @@
+
+  beforeEach( waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        MockComponent({ selector: 'chef-button', inputs: ['disabled'] }),
+        MockComponent({ selector: 'chef-loading-spinner' }),
+        MockComponent({ selector: 'chef-form-field' }),
+        MockComponent({ selector: 'chef-error' }),
+        MockComponent({ selector: 'chef-toolbar' }),
+        MockComponent({ selector: 'app-infra-tab-change' }),
+        MockComponent({ selector: 'app-infra-tab', inputs: ['active', 'disabled'] }),
+
+        MockComponent({ selector: 'chef-modal',
+          inputs: ['visible']
+        }),
+        CreateEnvironmentModalComponent
+      ],
+      providers: [
+        { provide: TelemetryService, useClass: MockTelemetryService },
+        HttpClient, HttpHandler
+      ],
+      imports: [
+        ReactiveFormsModule,
+        StoreModule.forRoot(ngrxReducers, { runtimeChecks })
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateEnvironmentModalComponent);
+    component = fixture.componentInstance;
+    component.firstFormGroup = new FormBuilder().group({
+      name: ['', null],
+      description: ['', null]
+    });
+    component.thirdFormGroup = new FormBuilder().group({
+      dattr: ['', null]
+    });
+    component.fourthFormGroup = new FormBuilder().group({
+      oattr: ['', null]
+    });
+    component.openEvent = new EventEmitter();
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+
+  describe('create environment', () => {
+    let store: Store<NgrxStateAtom>;
+    const environment: Environment = {
+      org_id: 'chef_manage',
+      server_id: 'test',
+      name: 'test',
+      description: 'test environment',
+      cookbook_versions: {
+        aix: '> 2.3.4'
+      },
+      default_attributes: {
+        test: 'test'
+      },
+      override_attributes: {
+        test: 'test'
+      }
+    };
+
+    beforeEach(() => {
+      store = TestBed.inject(Store);
+    });
+
+    it('opening creat model and defaults tab to showing section', () => {
+      component.visible = true;
+      expect(component.detailsTab).toBe(true);
+      expect(component.constraintsTab).toBe(false);
+      expect(component.defaultTab).toBe(false);
+      expect(component.overrideTab).toBe(false);
+    });
+
+    it('opening create modal resets to empty string', () => {
+      component.visible = true;
+      expect(component.firstFormGroup.controls['name'].value).toEqual('');
+      expect(component.firstFormGroup.controls['description'].value).toEqual('');
+      expect(component.thirdFormGroup.controls['dattr'].value).toEqual('{}');
+      expect(component.fourthFormGroup.controls['oattr'].value).toEqual('{}');
+    });
+
+    it('on conflict error, modal remains open and displays conflict error', () => {
```

Comment:
This is not accurate. It is not validating "and displays conflict error".

Reply:
updated the testcase